todo_app/data/trello_items.py

```diff
@@ -1,88 +1,79 @@
-    return items
+def generate_params(params):
+    key = os.getenv('TRELLO_SECRET_KEY')
+    token = os.getenv('TRELLO_SECRET_TOKEN')
+    authentication_params = {'key': key, 'token': token}
+    return {**authentication_params, **params}
 
-def get_items_from_column(column_name):
+
+def get_all_trello_column_lists():
+    board_id = os.getenv('TRELLO_BOARD_ID')
     url = (f'https://api.trello.com/1/boards/{board_id}/lists')
 
-    response = requests.request(
-        'GET',
+    response = requests.get(
         url,
-        params={
-            'key': key,
-            'token': token,
-            'cards': 'open',
-            'fields': 'name'
-        }
+        params=generate_params({'cards': 'open',
+                                'fields': 'name'})
     )
 
-    for column_list in response.json():
-        print(column_list)
+    return response.json()
+
+
+def get_trello_column_list_from_name(column_name):
+    all_lists = get_all_trello_column_lists()
+
+    for column_list in all_lists:
         if column_list['name'] == column_name:
-            return convert_column_list_into_items(column_list)
+            return column_list
 
-    raise Exception(f'Could not find column with name {column_name} in this board')
+    raise Exception(
+        f'Could not find column with name {column_name} in this board')
 
-def get_column_id(column_name):
-    url = (f'https://api.trello.com/1/boards/{board_id}/lists')
 
-    response = requests.request(
-        'GET',
-        url,
-        params={
-            'key': key,
-            'token': token,
-            'cards': 'open',
-            'fields': 'name'
-        }
-    )
+def get_items_from_column(column_list):
+    items = []
+    for card in column_list['cards']:
+        items.append(Item.from_trello_card(card, column_list))
+    return items
+
+
+def get_all_items():
+    all_lists = get_all_trello_column_lists()
+    all_items = []
+    for column_list in all_lists:
+        all_items += get_items_from_column(column_list)
+
+    return all_items
+
+
+def get_column_id(column_name):
+    return get_trello_column_list_from_name(column_name)['id']
 
-    for column_list in response.json():
-        if column_list['name'] == column_name:
-            return column_list['id']
-    
-    raise Exception(f'Could not find column with name {column_name} in this board')
 
 def add_item_to_column(item, column_name):
     column_id = get_column_id(column_name)
 
     url = 'https://api.trello.com/1/cards'
 
-    requests.request(
-        'POST',
+    requests.post(
         url,
-        params= {
-            'key': key,
-            'token': token,
-            'name': item, 
-            'idList': column_id
-        }
+        params=generate_params({'name': item,
+                                'idList': column_id})
     )
 
+
 def move_item_to_column(item_id, column_name):
     column_id = get_column_id(column_name)
 
     url = (f'https://api.trello.com/1/cards/{item_id}')
```

Comment:
Being pedantic, this allows query string injection with e.g. `curl -X 'POST' 'http://localhost:5000/complete_item/60a4c79a6c47a13f6e02e02c%3Fdesc%3DSneaky_description'`

```suggestion
    url = (f'https://api.trello.com/1/cards/{urllib.parse.quote(item_id, safe="")}')
```

Reply:
Ah interesting, yep that makes sense thanks! :D